File: front-end/src/app/components/ressource/imprimante.ts
export class Imprimante {
    id: number;
    vitesse: string;
    resolution: string;
    codeBarre: string;
    dureeGarantie: string;
    dateLivraison : string;
    marque: string;
    
  
  
    constructor() {
      this.id = 0;
      this.codeBarre="";
      this.dateLivraison="";
      this.dureeGarantie="";
      this.resolution="";
      this.marque="";
      this.vitesse="";
     }
  }

File: Back-end/src/main/java/com/gestresmat/springjwt/models/Imprimante.java
package com.gestresmat.springjwt.models;

import javax.persistence.*;

import java.io.Serializable;
import java.util.Date;

@Entity
public class Imprimante extends Ressource implements Serializable {
    private String resolution;
    private String vitesse;


    public Imprimante(String codeBarre, String dureeGarantie, Date dateLivraison, String marque, String resolution, String vitesse) {
        super(codeBarre, dureeGarantie, dateLivraison, marque);
        this.resolution = resolution;
        this.vitesse = vitesse;
    }

    public Imprimante() {
        super();
    }

    public String getResolution() {
        return resolution;
    }

    public void setResolution(String resolution) {
        this.resolution = resolution;
    }

    public String getVitesse() {
        return vitesse;
    }

    public void setVitesse(String vitesse) {
        this.vitesse = vitesse;
    }
}